lxd/identity/cache.go

```diff
@@ -70,20 +70,20 @@ func (c *Cache) Get(authenticationMethod string, identifier string) (*CacheEntry
 }
 
 // GetByType returns a map of identifier to CacheEntry, where all entries have the given identity type.
-func (c *Cache) GetByType(identityType string) map[string]CacheEntry {
+func (c *Cache) GetByType(identityTypeStr string) map[string]CacheEntry {
 	c.mu.RLock()
 	defer c.mu.RUnlock()
 
 	// Explicitly ignore the error here. It is expected that the caller will use the constants defined in shared/api.
-	authenticationMethod, _ := AuthenticationMethodFromIdentityType(identityType)
-	entriesByAuthMethod, ok := c.entries[authenticationMethod]
+	identityType, _ := New(identityTypeStr)
```

Comment:
You need to test the error here to avoid a panic for invalid types on next line.